x-pack/plugins/apm/public/context/url_params_context/helpers.ts

```diff
@@ -26,13 +27,27 @@ export function getDateRange({
   rangeFrom?: string;
   rangeTo?: string;
 }) {
+  // If the previous state had the same range, just return that instead of calculating a new range.
   if (state.rangeFrom === rangeFrom && state.rangeTo === rangeTo) {
     return { start: state.start, end: state.end };
   }
 
+  const start = getParsedDate(rangeFrom);
+  const end = getParsedDate(rangeTo, { roundUp: true });
+
+  // `getParsedDate` will return undefined for invalid or empty dates. We return
+  // the previous state if either date is undefined.
+  if (!start || !end) {
+    return { start: state.start, end: state.end };
+  }
+
+  // Calculate ticks for the time ranges to produce nicely rounded values
+  const ticks = scaleUtc().domain([start, end]).nice().ticks();
```

Comment:
I didn't see a test that specifically mentioned the rounding. Did I just miss it?

Reply:
There isn't but I can add some.

Reply:
Please 🙏